helper/tests/codegen-110/src/main/raw-avro/by110/IntRecord.avsc

```diff
@@ -0,0 +1,19 @@
+{
+  "type": "record",
+  "namespace": "by110",
+  "name": "IntRecord",
+  "fields": [
+    {
+      "name": "unionField",
```

Comment:
A couple of suggestions to strengthen this before merge:

- Union demotion is covered (e.g. `LongRecord`), but consider adding:
  - **Overflow cases** (`long` > `Integer.MAX_VALUE` in a union) → should throw a clear `AvroTypeException`.
  - **Branch order differences** (writer `[long,null]` vs reader `[null,int]`) to confirm resolution is index-agnostic.
  - **Nested collections with unions** (array/map of `[null,long]` → `[null,int]`).

Reply:
Overflow and nested sure. I don't want to order branch order differences since that is an invalid schema evolution that we don't want to support.